### PR TITLE
drivers: watchdog: Migrate to new logging subsys

### PR DESCRIPTION
**Pull request by Olivier-ProGlove**
Migrate from `SYS_LOG` to `LOG` logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]
LGTM. 
I would rename commit title prefix to
drivers: watchdog: nrf: 
(nrf: is used for Nordic SOC-related patches, AFAICS)

File: drivers/watchdog/wdt_nrfx.c [hunk 1]
```diff
@@ -7,9 +7,10 @@
 #include <nrfx_wdt.h>
 #include <watchdog.h>
 
-#define SYS_LOG_DOMAIN "wdt_nrfx"
+#define LOG_MODULE_NAME wdt_nrfx
 #define SYS_LOG_LEVEL CONFIG_SYS_LOG_WDT_LEVEL
```
Comment (ydamigos) [inline drivers/watchdog/wdt_nrfx.c]: I opened similar PRs for stm32 I2C (#9385) and usb (#9384) drivers. I had to change this line to make it work with the new LOG subsys. Did it work without changing this line?

**Comment by anangl** [comment]
@ioannisg The commit title is okay as it is. This is a shim using the nrfx driver underneath, so it is perfectly fine to use the "nrfx:" prefix. Such prefix would be inappropriate for a commit changing something related to Nordic SoCs but not using the nrfx driver package.

**Comment by ydamigos** [comment]
@Olivier-ProGlove CONFIG_SYS_LOG_WDT_LEVEL depends on `SYS_LOG`. In order to work with the new `LOG` subsys you need to edit Kconfig file and change it to:
`depends on SYS_LOG || LOG`

**Comment by Olivier-ProGlove** [comment]
@ydamigos I have updated the pull-request to migrate all watchdog drivers to `LOG`

**Comment by Olivier-ProGlove** [comment]
I have rebased the pull-request to fix the conflict after this pull-request has been merged https://github.com/zephyrproject-rtos/zephyr/commit/8cf8db3a73aa0658d728006e93d9ef5e0361a1b2

**Review by galak — CHANGES_REQUESTED** [review]
Can you fix this for the change in LOG_MODULE_REGISTER to take the name as a param, and remove LOG_MODULE_NAME

**Comment by nashif** [comment]
cherrypicked in #10029

**Comment by Olivier-ProGlove** [comment]
@nashif That's a better cherry-pick compared to https://github.com/zephyrproject-rtos/zephyr/pull/9429#issuecomment-422175587 :+1:

**Comment by nashif** [comment]
addressed in #10029

**Pull request closed without merging**